curriculum/challenges/english/25-front-end-development/lecture-working-with-loops/6732c06654ea3442724284fe.md

```diff
@@ -8,7 +8,77 @@ dashedName: what-is-a-while-loop-and-how-does-it-differ-from-the-do-while-loop
 
 # --description--
 
-Watch the lecture video and answer the questions below.
+Watch the video or read the transcript and answer the questions below.
+
+# --transcript--
+
+What is a `while` loop and how does it differ from the `do...while` loop?
+
+In previous lecture videos, you learned how to work with `for` loops, `for...in` loops and `for...of` loops. In this lecture, you will learn about the `while` loop and the `do...while` loop.
+
+A `while` loop will run a block of code as long as the condition is true. Here is the basic syntax for a `while` loop:
+
+```js
+while (condition) {
+  // code block to be executed
+}
+```
+
+The condition is checked before the block of code is executed. If the condition is false, the block of code will not be executed.
+
+`while` loops are useful when you do not know how many times you need to run the block of code. Here is an example of using a `while` loop to validate user input:
+
+```js
+let userInput = prompt("Please enter a number between 1 and 10");
+
+while (isNaN(userInput) || Number(userInput) < 1 || Number(userInput) > 10) {
+  userInput = prompt("Invalid input. Please enter a number between 1 and 10.");
+}
+
+alert("You entered a valid number!");
+```
+
+In this example, we are using the `prompt` method to ask for user input. Then, the `while` loop first checks if the `userInput` is `NaN`. Remember that `NaN` is "Not a Number".  `isNaN` checks whether the value is `NaN` and that `NaN` is the result of an invalid number conversion.
+
+So if the user types in random characters or nothing at all, then they will be prompted with the message again. The `while` loop is also checking if the `userInput`, when converted to a number, is between `1` and `10`. 
+
+We are using the `Number` constructor here because prompt will return a string and we want only want to be explicitly checking numbers here.
```

Comment:
```suggestion
We are using the `Number` constructor here because a prompt will return a string and we want only want to be explicitly checking numbers here.
```